pkg/operator/ceph/cluster/watcher.go

```diff
@@ -65,12 +71,46 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
 	if !ok {
 		return false
 	}
+
+	// Get CephCluster
+	cluster := c.getCephCluster()
+
+	nodeLoss, err := k8sutil.GetOperatorSetting(ctx, c.context.Clientset, opcontroller.OperatorSettingConfigMapName, "ROOK_WATCH_FOR_NODE_FAILURE", "true")
+	if err != nil {
+		logger.Errorf("failed to get configmap value `ROOK_WATCH_FOR_NODE_FAILURE`. %v", err)
+		return false
+	}
+
+	var nodeWithTaint corev1.Node
```

Comment:
How about a boolean instead of assigning the node?
```suggestion
	nodeHasOutOfServiceTaint := false
```